core/arch/arm/plat-imx/imx-common.c

```diff
@@ -13,34 +13,43 @@
 #include <mm/core_memprot.h>
 #include <platform_config.h>
 
-static uint32_t imx_digproc(void)
+static int imx_cpu_type = (-1);
+static int imx_soc_revision = (-1);
+
+static void imx_digproc(void)
 {
-	static uint32_t reg;
+	uint32_t digprog;
 	vaddr_t anatop_addr;
 
-	if (!reg) {
-		anatop_addr = core_mmu_get_va(ANATOP_BASE, MEM_AREA_IO_SEC);
+	anatop_addr = core_mmu_get_va(ANATOP_BASE, MEM_AREA_IO_SEC);
+
+	if (!anatop_addr)
+		return;
+
+	digprog = io_read32(anatop_addr + DIGPROG_OFFSET);
 
-#if defined(CFG_MX7)
-		reg = io_read32(anatop_addr + OFFSET_DIGPROG_IMX7D);
-#elif defined(CFG_MX6SL)
-		reg = io_read32(anatop_addr + OFFSET_DIGPROG_IMX6SL);
-#else
-		reg = io_read32(anatop_addr + OFFSET_DIGPROG);
-#endif
-	}
+	/* Set the CPU type */
+	imx_cpu_type = ((digprog >> 16) & 0xFF);
 
-	return reg;
+	/* Set the SOC revision = (Major + 1).(Minor) */
+	imx_soc_revision =
+		(((digprog & 0xFF00) >> 4) + 0x10) | (digprog & 0x0F);
 }
 
 static uint32_t imx_soc_rev_major(void)
 {
-	return ((imx_digproc() & 0xff00) >> 8) + 1;
+	if (imx_soc_revision < 0)
```

Comment:
these test mandates major < 0x80. Maybe add an inline comment for this.

Reply:
Hi @etienne-lms , I don't get what you want me to comment here.

Reply:
sorry, it's a 32bit, large enough for the stored value.
Discard my comment :)

Reply:
Ok :)